src/material-examples/autocomplete-display/autocomplete-display-example.ts

```diff
@@ -31,8 +31,8 @@ export class AutocompleteDisplayExample {
   ngOnInit() {
     this.filteredOptions = this.myControl.valueChanges
       .pipe(
-        startWith({} as User),
-        map(user => user && typeof user === 'object' ? user.name : user),
+        startWith<string | User>(this.myControl.value),
+        map(value => value && typeof value === 'object' ? value.name : value as string),
```

Comment:
`as` was necessary to run properly on Stackblitz.

Reply:
The `typeof something === 'object'` is a little tricky since it'll also be true for `null`. This might be nicer if you changed it to: 

```ts
return typeof value === 'string' ? value : value.name;
```

Reply:
*"The typeof something === 'object' is a little tricky since it'll also be true for null"*

In this case, it won't happen because the check is `value && typeof value === 'object'`. Anyway, I changed it since it's more readable.